document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -1977,8 +1982,201 @@ public void testAtomicUpdateDocumentWithoutSelections(final String datastoreName
+              .subDocumentValue(SubDocumentValue.of(new Integer[] {8, 2}))
+              .build();
+      final SubDocumentUpdate remove_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.planets")
+              .operator(REMOVE_ALL_FROM_LIST)
+              .subDocumentValue(SubDocumentValue.of(new String[] {"Pluto", "Mars"}))
+              .build();
+      final List<SubDocumentUpdate> new_updates =
+          List.of(set_new, unset_new, add_new, append_new, remove_new);
+
+      final CloseableIterator<Document> iterator_new =
+          collection.bulkUpdate(
+              query, new_updates, UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build());
+      assertDocsAndSizeEqualWithoutOrder(
+          datastoreName, iterator_new, "query/update_operator/updated2.json", 9);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(AllProvider.class)
+    void testRemoveAllOccurrancesFromIntegerList(final String datastoreName) throws IOException {
+      final Collection collection = getCollection(datastoreName, UPDATABLE_COLLECTION_NAME);
+      createCollectionData("query/updatable_collection_data.json", UPDATABLE_COLLECTION_NAME);
+
+      final SubDocumentUpdate unset =
+          SubDocumentUpdate.builder().subDocument("sales").operator(UNSET).build();
+      final SubDocumentUpdate add =
+          SubDocumentUpdate.builder()
+              .subDocument("props.added.list")
+              .subDocumentValue(SubDocumentValue.of(new Integer[] {5, 1, 5}))
+              .build();
+
+      final Query query = Query.builder().build();
+      final List<SubDocumentUpdate> updates = List.of(add, unset);
+
+      final CloseableIterator<Document> iterator =
+          collection.bulkUpdate(
+              query, updates, UpdateOptions.builder().returnDocumentType(NONE).build());
+
+      final SubDocumentUpdate remove =
+          SubDocumentUpdate.builder()
+              .subDocument("props.added.list")
+              .operator(REMOVE_ALL_FROM_LIST)
+              .subDocumentValue(SubDocumentValue.of(5))
+              .build();
+      final List<SubDocumentUpdate> new_updates = List.of(remove);
+
+      final CloseableIterator<Document> iterator_new =
+          collection.bulkUpdate(
+              query, new_updates, UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build());
+      assertDocsAndSizeEqualWithoutOrder(
+          datastoreName,
+          iterator_new,
+          "query/update_operator/updated_removes_all_occurrances_from_integer_list.json",
+          9);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(AllProvider.class)
+    void testAddToListIfAbsentDoesNotDeduplicateTheExistingList(final String datastoreName)
+        throws IOException {
+      final Collection collection = getCollection(datastoreName, UPDATABLE_COLLECTION_NAME);
+      createCollectionData("query/updatable_collection_data.json", UPDATABLE_COLLECTION_NAME);
+
+      final SubDocumentUpdate unset =
```

Comment:
Do we need `unset` on `sales` for the tests we are doing - or is this just to reduce JSON file size? I see the same condition in the previous test too so was wondering - is it required?

Reply:
> is this just to reduce JSON file size?

Yes. That's the idea. Otherwise not required.